rfcs/0007-filtering.md

```diff
@@ -0,0 +1,211 @@
+# RFC Proposal
+[RFC Proposal]: #rfc-proposal
+
+* RFC Proposal ID: `0007-filtering-fields`
+* Start Date: 2020-08-21
+* RFC PR: [rfcs#0007](https://github.com/CMSgov/beneficiary-fhir-data/pull/345)
+* JIRA Ticket(s):
+    * [https://jira.cms.gov/browse/AB2D-1863](https://jira.cms.gov/browse/AB2D-1863)
+
+This proposal will suggest adding server side filtering to API calls that return EOBs to the client in order to 
+reduce on the amount of data that needs to be transferred. 
+
+## Table of Contents
+[Table of Contents]: #table-of-contents
+
+* [RFC Proposal](#rfc-proposal)
+* [Table of Contents](#table-of-contents)
+* [Motivation](#motivation)
+* [Proposed Solution](#proposed-solution)
+    * [Proposed Solution: Detailed Design](#proposed-solution-detailed-design)
+    * [Proposed Solution: Unresolved Questions](#proposed-solution-unresolved-questions)
+    * [Proposed Solution: Drawbacks](#proposed-solution-drawbacks)
+    * [Proposed Solution: Notable Alternatives](#proposed-solution-notable-alternatives)
+* [Future Possibilities](#future-possibilities)
+
+## Motivation
+[Motivation]: #motivation
+
+AB2D receives more data than needed from BFD when we make API calls to gather EOBs (Explanation of Benefits).  AB2D
+currently filters out unrelevant fields.  This might not be terrible for dozens of API calls, but for 
+10 to 100 thousand requests it can impact our performance significantly.  Since we only need a projection of the data
+(10 out of 69 top level getters; 6 out of 42 fields in the ItemComponent), it makes
+sense for the server to send us only the data consumed.  This reduces bandwidth requirements for receiving the data.
```

Comment:
I totally buy this argument but it'd still be good to back it up with evidence. For example: "Of the total time AB2D spends on EOB requests, XX% of this time is currently spent on data filtering, which could instead be handled much faster by BFD." (And fill in the blank.)